doc/source/dev/core-dev/deprecations.rst.inc

```diff
@@ -7,25 +7,47 @@ There are various reasons for wanting to remove existing functionality: it's
 buggy, the API isn't understandable, it's superseded by functionality with
 better performance, it needs to be moved to another SciPy submodule, etc.
 
-In general it's not a good idea to remove something without warning users about
-that removal first.  Therefore this is what should be done before removing
+In general, it's not a good idea to remove something or make other backwards
+compatibility breaking changes without warning users about
+the change first. Therefore this is what should be done before deprecating
 something from the public API:
 
 #. Propose to deprecate the functionality on the scipy-dev mailing list and get
    agreement that that's OK.
-#. Add a ``DeprecationWarning`` for it, which states that the functionality was
-   deprecated, and in which release. For Cython APIs, see
+#. When changing a function signature (e.g. deprecating a keyword argument), this
+   opportunity should be used to convert that function to keyword-only usage,
+   see `GH-14714<https://github.com/scipy/scipy/issues/14714>`_.
```

Comment:
Actually https://docs.scipy.org/doc/scipy/dev/missing-bits.html#required-keyword-names is a better reference here than the issue (thanks for digging up the reference!). Would you mind updating the link?